api/common.go

```diff
@@ -16,22 +16,33 @@ import (
 
 // Common constants for daemon and client.
 const (
-	// Current REST API version
+	// Version of Current REST API
 	Version version.Version = "1.20"
 
-	// Minimun REST API version supported
+	// MinVersion represents Minimun REST API version supported
 	MinVersion version.Version = "1.12"
 
-	// Default filename with Docker commands, read by docker build
+	// DefaultDockerfileName is the Default filename with Docker commands, read by docker build
 	DefaultDockerfileName string = "Dockerfile"
 )
 
-type ByPrivatePort []types.Port
+// byPrivatePort is a temporary type used for sort with sort.Sort() func.
+// and sort.Sort() will make one call to data.Len to determine n, and O(n*log(n)) calls
+// to data.Less and data.Swap.
+type byPrivatePort []types.Port
 
-func (r ByPrivatePort) Len() int           { return len(r) }
-func (r ByPrivatePort) Swap(i, j int)      { r[i], r[j] = r[j], r[i] }
-func (r ByPrivatePort) Less(i, j int) bool { return r[i].PrivatePort < r[j].PrivatePort }
+// Len returns length of Port array, it's a customized Len() func used by sort.Sort()
```

Comment:
This actually wasn't golint warnings. I think it's enough to write in type docstring, that that type implements `sort.Interface`